vllm/config.py

```diff
@@ -129,6 +129,7 @@ def __init__(
         skip_tokenizer_init: bool = False,
         served_model_name: Optional[Union[str, List[str]]] = None,
         multimodal_config: Optional["MultiModalConfig"] = None,
+        use_output_proc_callback: Optional[bool] = False,
```

Comment:
I think @WoosukKwon proposal to make the async callback default is a good direction, so we can see the benefits already from this PR. You can enable the callback by default and disable it only for the cases where it does not work currently (beam search, etc...). Follow up PRs can add the missing features.